Sources/Utility/which.swift

```diff
@@ -10,10 +10,15 @@
 
 import Foundation
 
+private var memoized = [String: String]()
+
 public func which(_ arg0: String) -> String {
     if arg0.isAbsolute {
         return arg0
+    } else if let fullpath = memoized[arg0] {
+        return fullpath
     } else if let fullpath = try? Utility.popen(["/bin/sh", "-c", "which \(arg0)"]) {
+        memoized[arg0] = fullpath.chomp()
```

Comment:
We should probably just implement the lookup directly rather than actually shell out for this relatively trivial task.